File: 백준/Silver/1639. 행운의 티켓/행운의 티켓.py
import sys
input = sys.stdin.readline


def solve(N: int, S: str) -> int:
    rtn = 0

    for n in range(N):
        left, right = 0, 0
        i, j = n, n + 1
        while i >= 0 and j < N:
            left += int(S[i])
            right += int(S[j])
            if left == right:
                rtn = max(rtn, j - i + 1)
            i -= 1
            j += 1
        
    return rtn


def main():
    # 입력값
    S = input().rstrip()

    # 풀이
    result: list[int] = solve(len(S), S)
    
    # 출력
    print(result)

if __name__ == "__main__":
    main()

File: 백준/Silver/1639. 행운의 티켓/행운의 티켓.cc
#include <iostream>

using namespace std;

int solve(string s)
{
  int N = s.length();
  int result = 0;

  for (int n = 0; n < N; n++)
  {
    int left = 0, right = 0;
    int i = n, j = i+1;
    while (i >= 0 && j < N)
    {
      left += int(s[i]);
      right += int(s[j]);
      if (left == right && (j - i + 1) > result)
        result = j-i+1;
      i--;
      j++;
    }
  }
  return result;
}

int main()
{
  string s;
  cin >> s;

  int result = solve(s);
  cout << result << endl;

  return 0;
}
